sdk/src/main/java/com/horizen/account/api/rpc/types/TransactionArgs.java

```diff
@@ -35,19 +39,75 @@ public byte[] getData() {
         return Numeric.hexStringToByteArray(hex);
     }
 
+    /**
+     * Set sender address or use zero address if none specified.
+     */
     public byte[] getFrom() {
         return from == null ? new byte[Address.LENGTH] : from.toBytes();
     }
 
-    public Message toMessage() {
+    /**
+     * Converts the transaction arguments to the Message type used by the core.
+     * This method is used in calls and traces that do not require a real live transaction.
+     * Reimplementation of the same logic in GETH.
+     */
+    public Message toMessage(BigInteger baseFee) throws RpcException {
+        if (gasPrice != null && (maxFeePerGas != null || maxPriorityFeePerGas != null)) {
+            throw new RpcException(RpcError.fromCode(
+                    RpcCode.InvalidParams,
+                    "both gasPrice and (maxFeePerGas or maxPriorityFeePerGas) specified"
+            ));
+        }
+        // global RPC gas cap (in geth this is a config variable)
+        var gasLimit = BigInteger.valueOf(50_000_000);
```

Comment:
This is just for RPC calls to prevent `eth_calls` with very high gas limits, not relevant for the network. Tbh I don't understand why this is 50M with the block limit being at 30M :shrug:

Reply:
Thinking about it, `eth_call` can call getters that are never used within transactions, and these could theoretically require more gas than would be allowed in a single transaction.. :thinking:

Reply:
Fine, let's keep it as a constant as well near to the 30M constant and add some description how these values are related.